src/jit/codegenarmarch.cpp

```diff
@@ -880,6 +886,216 @@ void CodeGen::genPutArgReg(GenTreeOp* tree)
     genProduceReg(tree);
 }
 
+#ifdef _TARGET_ARM_
+//---------------------------------------------------------------------
+// genPutArgStruct - generate code for a GT_PUTARG_STRUCT node
+//
+// Arguments
+//    tree - the GT_PUTARG_STRUCT node
+//
+// Return value:
+//    None
+//
+void CodeGen::genPutArgSplit(GenTreePutArgSplit* treeNode)
+{
+    assert(treeNode->OperIs(GT_PUTARG_SPLIT));
+
+    GenTreePtr source = treeNode->gtOp1;
+    emitter*   emit   = getEmitter();
+
+    noway_assert((source->OperGet() == GT_LCL_VAR) || (source->OperGet() == GT_OBJ));
+
+    var_types targetType = source->TypeGet();
+    noway_assert(varTypeIsStruct(targetType));
+
+    regNumber baseReg = treeNode->ExtractTempReg();
+    regNumber addrReg = REG_NA;
+
+    GenTreeLclVarCommon* varNode  = nullptr;
+    GenTreePtr           addrNode = nullptr;
+
+    if (source->OperGet() == GT_LCL_VAR)
+    {
+        varNode = source->AsLclVarCommon();
+    }
+    else
+    {
+        assert(source->OperGet() == GT_OBJ);
+
+        addrNode = source->gtOp.gtOp1;
+
+        // addrNode can either be a GT_LCL_VAR_ADDR or an address expression
+        //
+        if (addrNode->OperGet() == GT_LCL_VAR_ADDR)
+        {
+            // We have a GT_OBJ(GT_LCL_VAR_ADDR)
+            //
+            // We will treat this case the same as above
+            // (i.e if we just had this GT_LCL_VAR directly as the source)
+            // so update 'source' to point this GT_LCL_VAR_ADDR node
+            // and continue to the codegen for the LCL_VAR node below
+            //
+            varNode  = addrNode->AsLclVarCommon();
+            addrNode = nullptr;
+        }
+    }
+
+    // Either varNode or addrNOde must have been setup above,
+    // the xor ensures that only one of the two is setup, not both
+    assert((varNode != nullptr) ^ (addrNode != nullptr));
+
+    BYTE* gcPtrs = nullptr;
+
+    unsigned gcPtrCount; // The count of GC pointers in the struct
+    int      structSize;
+    bool     isHfa;
```

Comment:
You should be able to get these values from the node. In inherits `gtNumberReferenceSlots` and `getArgSize()`, and it would make sense to add an `isHfa` flag (probably to `GenTreePutArgStk` to avoid extra padding).